flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -470,6 +472,34 @@ public void testCassandraTableSink() throws Exception {
 		Assert.assertTrue("The input data was not completely written to Cassandra", input.isEmpty());
 	}
 
+	@Test
+	public void testCassandraBatchPojoFormat() throws Exception {
+
+		OutputFormat<Tuple3<String, Integer, Integer>> sink = new CassandraTupleOutputFormat<>(injectTableName(INSERT_DATA_QUERY), builder);
+		sink.configure(new Configuration());
+		sink.open(0, 1);
+
+		for (Tuple3<String, Integer, Integer> value : collection) {
+			sink.writeRecord(value);
+		}
+
+		sink.close();
+
+		InputFormat<CustomCassandraAnnotatedPojo, InputSplit> source = new CassandraPojoInputFormat<>(injectTableName(SELECT_DATA_QUERY), builder, CustomCassandraAnnotatedPojo.class);
+		source.configure(new Configuration());
+		source.open(null);
+
+		List<CustomCassandraAnnotatedPojo> result = new ArrayList<>();
+
+		while (!source.reachedEnd()) {
+			CustomCassandraAnnotatedPojo temp = source.nextRecord(new CustomCassandraAnnotatedPojo());
+			result.add(temp);
+		}
+
+		source.close();
+		Assert.assertEquals(20, result.size());
```

Comment:
Can we also convert `collection` of tuples into list of `CustomCassandraAnnotatedPojo` and check the contents as well, additionally to size?

Reply:
To save easily a list of CustomCassandraAnnotatedPojo, i used the CassandraPojoSink

Reply:
maybe example would be also more consistent if we use `CassandraPojoSink` there as well to populate the table, instead of raw tuples.